### PR TITLE
refactor to use new collection classes

### LINKED ISSUE #880: refactoring: use of cl_object_collection_iterator
object `cl_object_collection_iterator` is not a steampunk relasesed object #632

suggest implementing the iterator functionality in a new `zcl_excel_iterator` class, and using that instead

### PR DESCRIPTION
**Pull request by larshp**
closes #880

### CODE REVIEW

**Review by sandraros — CHANGES_REQUESTED** [review]
LGTM but class names in CLASS IMPLEMENTATION are all changed to upper case. Is it normal?

**Comment by larshp** [comment]
its changed by SE80, 

if you'd like to enforce a specific style, suggest enabling https://rules.abaplint.org/keyword_case/, https://github.com/abap2xlsx/abap2xlsx/blob/master/abaplint.json#L109

File: src/zcl_excel.clas.abap [hunk 1]
```diff
@@ -184,7 +184,7 @@ ENDCLASS.
 
 
 
-CLASS zcl_excel IMPLEMENTATION.
+CLASS ZCL_EXCEL IMPLEMENTATION.
```
Comment (sandraros) [inline src/zcl_excel.clas.abap]:
```suggestion
CLASS zcl_excel IMPLEMENTATION.
```

File: src/zcl_excel_columns.clas.abap [hunk 2]
```diff
@@ -41,13 +41,13 @@ CLASS zcl_excel_columns DEFINITION
...
 
-    DATA columns TYPE REF TO cl_object_collection .
+    DATA columns TYPE REF TO zcl_excel_collection .
     DATA columns_hashed TYPE mty_ts_hashed_column .
 ENDCLASS.
 
 
 
-CLASS zcl_excel_columns IMPLEMENTATION.
+CLASS ZCL_EXCEL_COLUMNS IMPLEMENTATION.
```
Comment (sandraros) [inline src/zcl_excel_columns.clas.abap]:
```suggestion
CLASS zcl_excel_columns IMPLEMENTATION.
```

File: src/zcl_excel_comments.clas.abap [hunk 3]
```diff
@@ -33,12 +33,12 @@ CLASS zcl_excel_comments DEFINITION
...
   PRIVATE SECTION.
 
-    DATA comments TYPE REF TO cl_object_collection .
+    DATA comments TYPE REF TO zcl_excel_collection .
 ENDCLASS.
 
 
 
-CLASS zcl_excel_comments IMPLEMENTATION.
+CLASS ZCL_EXCEL_COMMENTS IMPLEMENTATION.
```
Comment (sandraros) [inline src/zcl_excel_comments.clas.abap]:
```suggestion
CLASS zcl_excel_comments IMPLEMENTATION.
```

File: src/zcl_excel_data_validations.clas.abap [hunk 4]
```diff
@@ -32,12 +32,12 @@ CLASS zcl_excel_data_validations DEFINITION
...
   PRIVATE SECTION.
 
-    DATA data_validations TYPE REF TO cl_object_collection .
+    DATA data_validations TYPE REF TO zcl_excel_collection .
 ENDCLASS.
 
 
 
-CLASS zcl_excel_data_validations IMPLEMENTATION.
+CLASS ZCL_EXCEL_DATA_VALIDATIONS IMPLEMENTATION.
```
Comment (sandraros) [inline src/zcl_excel_data_validations.clas.abap]:
```suggestion
CLASS zcl_excel_data_validations IMPLEMENTATION.
```

File: src/zcl_excel_drawings.clas.abap [hunk 5]
```diff
@@ -48,12 +48,12 @@ CLASS zcl_excel_drawings DEFINITION
...
 *"* private components of class ZCL_EXCEL_DRAWINGS
 *"* do not include other source files here!!!
-    DATA drawings TYPE REF TO cl_object_collection .
+    DATA drawings TYPE REF TO zcl_excel_collection .
 ENDCLASS.
 
 
 
-CLASS zcl_excel_drawings IMPLEMENTATION.
+CLASS ZCL_EXCEL_DRAWINGS IMPLEMENTATION.
```
Comment (sandraros) [inline src/zcl_excel_drawings.clas.abap]:
```suggestion
CLASS zcl_excel_drawings IMPLEMENTATION.
```

File: src/zcl_excel_fill_template.clas.abap [hunk 6]
```diff
@@ -86,7 +86,7 @@ ENDCLASS.
 
 
 
-CLASS zcl_excel_fill_template IMPLEMENTATION.
+CLASS ZCL_EXCEL_FILL_TEMPLATE IMPLEMENTATION.
```
Comment (sandraros) [inline src/zcl_excel_fill_template.clas.abap]:
```suggestion
CLASS zcl_excel_fill_template IMPLEMENTATION.
```

File: src/zcl_excel_ranges.clas.abap [hunk 7]
```diff
@@ -36,12 +36,12 @@ CLASS zcl_excel_ranges DEFINITION
...
   PRIVATE SECTION.
 
-    DATA ranges TYPE REF TO cl_object_collection .
+    DATA ranges TYPE REF TO zcl_excel_collection .
 ENDCLASS.
 
 
 
-CLASS zcl_excel_ranges IMPLEMENTATION.
+CLASS ZCL_EXCEL_RANGES IMPLEMENTATION.
```
Comment (sandraros) [inline src/zcl_excel_ranges.clas.abap]:
```suggestion
CLASS zcl_excel_ranges IMPLEMENTATION.
```

File: src/zcl_excel_reader_2007.clas.abap [hunk 8]
```diff
@@ -311,7 +311,7 @@ ENDCLASS.
 
 
 
-CLASS zcl_excel_reader_2007 IMPLEMENTATION.
+CLASS ZCL_EXCEL_READER_2007 IMPLEMENTATION.
```
Comment (sandraros) [inline src/zcl_excel_reader_2007.clas.abap]:
```suggestion
CLASS zcl_excel_reader_2007 IMPLEMENTATION.
```

File: src/zcl_excel_rows.clas.abap [hunk 9]
```diff
@@ -52,14 +52,14 @@ CLASS zcl_excel_rows DEFINITION
...
-    DATA rows TYPE REF TO cl_object_collection .
+    DATA rows TYPE REF TO zcl_excel_collection .
     DATA rows_hashed TYPE mty_ts_hashed_row .
 
 ENDCLASS.
 
 
 
-CLASS zcl_excel_rows IMPLEMENTATION.
+CLASS ZCL_EXCEL_ROWS IMPLEMENTATION.
```
Comment (sandraros) [inline src/zcl_excel_rows.clas.abap]:
```suggestion
CLASS zcl_excel_rows IMPLEMENTATION.
```

File: src/zcl_excel_styles.clas.abap [hunk 10]
```diff
@@ -43,12 +43,12 @@ CLASS zcl_excel_styles DEFINITION
...
   PRIVATE SECTION.
 
-    DATA styles TYPE REF TO cl_object_collection .
+    DATA styles TYPE REF TO zcl_excel_collection .
 ENDCLASS.
 
 
 
-CLASS zcl_excel_styles IMPLEMENTATION.
+CLASS ZCL_EXCEL_STYLES IMPLEMENTATION.
```
Comment (sandraros) [inline src/zcl_excel_styles.clas.abap]:
```suggestion
CLASS zcl_excel_styles IMPLEMENTATION.
```

File: src/zcl_excel_styles_cond.clas.abap [hunk 11]
```diff
@@ -36,12 +36,12 @@ CLASS zcl_excel_styles_cond DEFINITION
...
   PRIVATE SECTION.
 
-    DATA styles_cond TYPE REF TO cl_object_collection .
+    DATA styles_cond TYPE REF TO zcl_excel_collection .
 ENDCLASS.
 
 
 
-CLASS zcl_excel_styles_cond IMPLEMENTATION.
+CLASS ZCL_EXCEL_STYLES_COND IMPLEMENTATION.
```
Comment (sandraros) [inline src/zcl_excel_styles_cond.clas.abap]:
```suggestion
CLASS zcl_excel_styles_cond IMPLEMENTATION.
```

File: src/zcl_excel_worksheet.clas.abap [hunk 12]
```diff
@@ -735,7 +741,7 @@ ENDCLASS.
 
 
 
-CLASS zcl_excel_worksheet IMPLEMENTATION.
+CLASS ZCL_EXCEL_WORKSHEET IMPLEMENTATION.
```
Comment (sandraros) [inline src/zcl_excel_worksheet.clas.abap]:
```suggestion
CLASS zcl_excel_worksheet IMPLEMENTATION.
```

File: src/zcl_excel_worksheets.clas.abap [hunk 13]
```diff
@@ -39,12 +39,12 @@ CLASS zcl_excel_worksheets DEFINITION
...
   PRIVATE SECTION.
 
-    DATA worksheets TYPE REF TO cl_object_collection .
+    DATA worksheets TYPE REF TO zcl_excel_collection .
 ENDCLASS.
 
 
 
-CLASS zcl_excel_worksheets IMPLEMENTATION.
+CLASS ZCL_EXCEL_WORKSHEETS IMPLEMENTATION.
```
Comment (sandraros) [inline src/zcl_excel_worksheets.clas.abap]:
```suggestion
CLASS zcl_excel_worksheets IMPLEMENTATION.
```

File: src/zcl_excel_writer_csv.clas.abap [hunk 14]
```diff
@@ -53,7 +53,7 @@ ENDCLASS.
 
 
 
-CLASS zcl_excel_writer_csv IMPLEMENTATION.
+CLASS ZCL_EXCEL_WRITER_CSV IMPLEMENTATION.
```
Comment (sandraros) [inline src/zcl_excel_writer_csv.clas.abap]:
```suggestion
CLASS zcl_excel_writer_csv IMPLEMENTATION.
```

File: src/zcl_excel_writer_huge_file.clas.abap [hunk 15]
```diff
@@ -35,7 +35,7 @@ ENDCLASS.
 
 
 
-CLASS zcl_excel_writer_huge_file IMPLEMENTATION.
+CLASS ZCL_EXCEL_WRITER_HUGE_FILE IMPLEMENTATION.
```
Comment (sandraros) [inline src/zcl_excel_writer_huge_file.clas.abap]:
```suggestion
CLASS zcl_excel_writer_huge_file IMPLEMENTATION.
```

File: src/zcl_excel_writer_xlsm.clas.abap [hunk 16]
```diff
@@ -31,7 +31,7 @@ ENDCLASS.
 
 
 
-CLASS zcl_excel_writer_xlsm IMPLEMENTATION.
+CLASS ZCL_EXCEL_WRITER_XLSM IMPLEMENTATION.
```
Comment (sandraros) [inline src/zcl_excel_writer_xlsm.clas.abap]:
```suggestion
CLASS zcl_excel_writer_xlsm IMPLEMENTATION.
```

File: src/zcl_excel_writer_2007.clas.abap [hunk 17]
```diff
@@ -265,11 +265,7 @@ ENDCLASS.
 
 
 
-CLASS zcl_excel_writer_2007 IMPLEMENTATION.
-
-  METHOD constructor.
-    me->ixml = cl_ixml=>create( ).
-  ENDMETHOD.
+CLASS ZCL_EXCEL_WRITER_2007 IMPLEMENTATION.
```
Comment (sandraros) [inline src/zcl_excel_writer_2007.clas.abap]:
```suggestion
CLASS zcl_excel_writer_2007 IMPLEMENTATION.
```

**Review by sandraros — CHANGES_REQUESTED** [review]
Changing back to `CLASS zcl_<alllowercase> IMPLEMENTATION`. Proposed lint rule in #941.

**Comment by larshp** [comment]
reopen?

**Pull request merged**